### ISSUE TITLE
UI: Allow users to submit incorrect k8s pattern names

### ISSUE THREAD

**Issue by fivenp** [kubeshop/testkube#3121]
**Describe the enhancement you'd like to see**
Right now we throw errors whenever a user tries to enter an invalid `name` for a resource (we use names as IDs so we want them to be k8s conform). 

The errors we throw are very different. Sometimes we do Frontendvalidation for k8s patterns, sometimes the backend throws an error: 

![image](https://user-images.githubusercontent.com/132332/215476179-e4f7f585-7532-4829-93f4-72ef71128d8f.png)

![](https://user-images.githubusercontent.com/132332/215464877-564be4b1-4fad-4dc1-831f-28d832752c6a.png)

### Solution

Let's allow the user to enter whatever he wants as a name and convert that name to a working k8s pattern **ON SAVE** behind the scenes for them (e.g. use a slug generator with some rules) 

The main goal is to never have a form submit fail because of a wrong "name" :) 

### Resource name's we should cover: 

* Test creation - Test name
* Test Suite creation - Test Suite name
* Source creation – Source name
* Custom executor creation – Executor name

**Comment by fivenp** [comment]
we can de-prio this very much for now

**Comment by pavloburchak** [comment]
@fivenp any update on this one?

**Comment by fivenp** [comment]
no... we have way more important things to work on right now – let's move this even lower :)
